### PR TITLE
fix #280829 spanner-drop-apply.tour

### PR DESCRIPTION
**Pull request by ericfont**
Added new tour message triggered when either dropping or double-clicking spanners from the palette.

Explains how to extend a spanner's range with Shift-Right/Left.
Explains how to add a new spanner across a pre-selected range.

### CODE REVIEW

File: mscore/dragdrop.cpp [hunk 1]
```diff
@@ -548,6 +549,10 @@ void ScoreView::dropEvent(QDropEvent* event)
                         delete editData.dropElement;
                         break;
                   }
+            
+            if (editData.dropElement->isSpanner())
+                  TourHandler::startTour("spanner-drop-apply");
```
[the remark this answers is not in the document]
Comment (ericfont) [reply mscore/dragdrop.cpp]: I don't think I can.

**Comment by JoshuaBonn1** [comment]
Should there be a highlighted widget? I've seen scoreview used for a majority of score related items. 

Also, a consistency should be reached on the file naming "tour-name.tour" or "tourname.tour" (I like the first one) Not necessary for this, but just to mention

**Comment by ericfont** [comment]
> before or after the line is rendered? I strongly think af

good point.  I guess after rendered would be better...  But that means I have to keep track of a bool triggered...

**Comment by ericfont** [comment]
standby for new push -f

**Comment by JoshuaBonn1** [comment]
> > before or after the line is rendered? I strongly think af
> 
> good point. I guess after rendered would be better... But that means I have to keep track of a bool triggered...

Do you mean something with the elements or with the tour? The tour keeps track of it anyways and it just needs to be placed after endcmd (at least I believe so)

**Comment by ericfont** [comment]
@JoshuaBonn1 I've updated the commit such that the score cmd completes (so rendering is finished) before the tour initiates...here's the new behavior:

volta-tour_

**Comment by JoshuaBonn1** [comment]
That's much better! I still think some highlighting would be nice. Maybe scoreview and the line palette? Others can comment on that.

**Comment by ericfont** [comment]
> I still think some highlighting would be nice. Maybe scoreview and the line palette? 

Naw.  If there was something to highlight, it would specifically be just the dropped spanner.  But I don't think the tour framework can do such specifics.

**Comment by JoshuaBonn1** [comment]
> > I still think some highlighting would be nice. Maybe scoreview and the line palette?
> 
> Naw. If there was something to highlight, it would specifically be just the dropped spanner. But I don't think the tour framework can do such specifics.

Yet ;-) I can't see anything else odd, so I'm done.

**Comment by ericfont** [comment]
thanks for your review.  I think this is good enough to be merged now.

If any changes to tour framework occur in future, such as specifically highlighting elements or guidelines for tour names or message text changes, can update this code with a new PR.

**Pull request merged**